### ISSUE TITLE
Assertion failed with merged bam files

### ISSUE THREAD

**Issue by xingpel** [freebayes/freebayes#481]
Hi Erik,
I'm using freebayes to call SNPs on a multisample bam file merged with picardtools while kept read group, and bam file has been sorted already.

Initially I encountered

`Segmentation fault (core dumped)`

after I set 'ulimit -s unlimited', it seems fixed the issue but later another issue poped up as assertion error after produced vcf file with only 2 hundreds of SNPs (where it should be 100-1000 times more). However, if I use non-merged bam files, it does not compain any error but the vcf file did not contain any SNPs at all. 

I'm using freebayes v1.2.0.

Below is my input code and error message:
```
user@TAR:/data$ freebayes -f ${genome} --min-base-quality 10 --min-supporting-allele-qsum 10 --read-mismatch-limit 3 --min-coverage 5 --no-indels --min-alternate-count 4 --exclude-unobserved-genotypes --genotype-qualities --ploidy 2 --no-mnps --no-complex --mismatch-base-quality-threshold 10 ${dat}/merged.sorted.bam > ${dat}/result.vcf 

freebayes: freebayes.cpp:628: int main(int, char**): Assertion `m != NULL' failed.
Aborted (core dumped)
```

Thank you in advance!

**Comment by xingpel** [comment]
A few more information from the debug mode:
```

snp:1X:1617252:1:T has support of 4 in individual P1-18GH-256-39 (4 obs) and fraction 1.00000
filtered genotype alleles
adding allele to result alleles G
adding allele to result alleles S:1617252:1X:A
adding allele to result alleles S:1617252:1X:T
built haplotype alleles, now there are 3 genotype alleles
reference:1M:1617252:1:G|snp:1X:1617252:1:A|snp:1X:1617252:1:T
genotype alleles: reference:1M:1617252:1:G|snp:1X:1617252:1:A|snp:1X:1617252:1:T
searching genotype space
freebayes: freebayes.cpp:628: int main(int, char**): Assertion `m != NULL' failed.
Aborted (core dumped)

```

**Comment by skurtis** [comment]
Hi Erik,

I'm getting the same issue when I merge individual BAM files into a combined BAM file using "samtools merge". freeBayes keeps giving me the error "int main(int, char**): Assertion `m != __null' failed." when running the combined file. But I do not get the error when I run individual BAM files. Is it possible that this represents an error with input BAM file itself regarding the alleles or genotype likelihoods? I think the error occurs during lines 620-636 in the freebayes.cpp file, but I'm unfamiliar with C++ and what parameters are awry.

Thanks,
Sarah

**Comment by tseemann** [comment]
How were the BAMs originally sorted?
Did you use `samtools merge` or `samtools collate`?
Do they have read groups assigned?

**Comment by skurtis** [comment]
Thanks for the reply! I used "samtools sort" to sort each individual BAM file by genomic coordinates and "samtools merge" to merge individual BAM files. Each individual BAM file had read groups assigned, which were maintained in the merge BAM file.
________________________________

**Comment by tseemann** [comment]
What does `freebayes --version` say?

**Comment by Andy-B-123** [comment]
Hi, I encountered the same error as above but found a strange solution. I provided the following commandline for analysing 48 individual samples mapped with BBCall and all containing unique ReadGroupID and SampleNames:
`freebayes --bam-list ListOfAllSamples.list --fasta-reference ../../../ref/NCBI_HaChr.clean.fasta --populations SamplePopulations.list --vcf ../../data/AllSamples/AllSamples.freebayes.vcf --ploidy 2 --use-best-n-alleles 100 --min-coverage 5 --exclude-unobserved-genotypes`
I initially also thought it was a problem with the use of multiple files but actually the error went away after I omitted the '--exclude-unobserved-genotypes' option. I notice in the original poster they also include the 'exclude-unobserved-genotype' flag. This might have been the cause of their problem too! I am running freebayes version  v1.3.2
